src/components/ItemFilters/FiltersContainer.tsx

```diff
@@ -17,62 +23,75 @@ interface ItemFilterContainerProps {
 }
 
 const ItemFilterContainer = ({ itemAggs }: ItemFilterContainerProps) => {
-  const { query } = useRouter()
-  const filterData = itemAggs.map((agg: ItemAggregation) => {
-    if (agg.field === "location") return new LocationFilterData(agg)
-    else return new ItemFilterData(agg)
-  })
-  const [activeFilters, setAppliedFilters] = useState({
-    location: [],
-    format: [],
-    status: [],
-  })
+  const router = useRouter()
+
+  const filterData = useRef(
```

Comment:
For type safety and to better see how `filterData` is structured, can you type this as `useRef<ItemFilterData[]>(`